### ISSUE TITLE
Cropland mask: Namibia

### ISSUE THREAD

**Issue by ivanzvonkov** [nasaharvest/crop-mask#218]
**Start year:** 2020
**Start month:** September

- [ ] Labeling project created
- [ ] Labeling completed
- [ ] Data added to repository
- [ ] Model trained
- [ ] Map made

**Comment by ivanzvonkov** [comment]
Low cropland amount, low metrics:
https://github.com/nasaharvest/crop-mask/blob/0b9c305ae71994b8d938ee34a77a295565ff7435/data/models.json#L19

Could be improved by NDVI stratification

**Comment by ivanzvonkov** [comment]
Next step adding WFP data

**Comment by ivanzvonkov** [comment]
@MsPixels now that you have some corrective points, the next step is to add them to the code base! Here's an example of me adding the the Namibia WFP data https://github.com/nasaharvest/crop-mask/pull/227
The **main** change is to datasets.py which you will have to update too!
More info here: https://github.com/nasaharvest/crop-mask#adding-new-labeled-data

**Comment by ivanzvonkov** [comment]
The compute cost above actually also accounts for Zambia v2 so it is actually lower

**Comment by MsPixels** [comment]
Yeah, @ivanzvonkov, major improvement indeed. Just wondering why the f1 score is this low despite this result.

**Comment by ivanzvonkov** [comment]
Additional data has been added in #243, now new model can be trained

**Comment by ivanzvonkov** [comment]
Model being trained https://github.com/nasaharvest/crop-mask/pull/250

**Comment by hannah-rae** [comment]
From Christina in meeting 1/19:
- 1500 crop type points with crop type and field boundary for season starting early 2023, use start month September still (collected Dec/Jan 2022-2023). Blake and Abena are cleaning that now.
- Goal: make in-season cropland map starting September 2022 through January 2023 which includes the 1500 field labels from above ( @MsPixels can create this?)
- Will be doing 2 more rounds of data collection in February (early season) and April/May (harvest stage) which can be used to evaluate the in-season map
- Re: making more representative samples using information that we have from FAO etc, Christina suggested also looking at sampling to match proportions of production historically in admin1 zones (e.g. so we sample more from zones producing more); there is also historical information in Crop Monitors that we could consider using
- @MsPixels can also look into making a map for WFP for cropland in 2021-2022 after the in-season map if time

**Comment by MsPixels** [comment]
Yes, @hannah-rae, I can create the in-season cropland map and hopefully the WFP map

**Comment by MsPixels** [comment]
> Model being trained #250

 Namibia_North_2020_v2
  
        "test_metrics": 
            "accuracy": 0.9324,
            "f1_score": 0.0625,
            "precision_score": 0.037,
            "recall_score": 0.2,
            "roc_auc_score": 0.8661

**Comment by MsPixels** [comment]
[Comparing V1 and V2 - Namibia_North](https://code.earthengine.google.com/37f8a203f57633bce5faa431b36dbda7)

Although this model predicts more crop fields than normal, it is able to correctly predict crop fields in the big yellow squares

**Comment by hannah-rae** [comment]
@MsPixels @ivanzvonkov  I was looking at the last comments on this issue but we don't have the "next steps" recorded for when it got picked back up. Are these the potential next steps?

- [ ] Re-create map with new model trained w/ corrected NDVI ( @ivanzvonkov I think you are re-training all models?)
- [ ] Have the ground truth data from Dec/Jan 2022-2023 already been added to the test set? 
- [ ] I have in an earlier comment that more ground data was to be collected in February 2023. @MsPixels have you heard from Christina or Blake if these new data are available?

Other things to try:
- [ ] Create stratified reference sample for validation/test instead of current random uniform
- [ ] Apply post-processing NDVI that @bhyeh is working on (may reduce false positives)
- [ ] Re-train model with high-quality data
- [ ] Data quality audit

**Comment by MsPixels** [comment]
Thanks Hannah for your feedback. Yes, these are potential next steps.

- The ground truth data from Dec/Jan 2022 - 2023 has been added to the test set
- Yes, February 2023 is being cleaned
-  I'll go ahead and generate a stratified sample for validation.

**Comment by MsPixels** [comment]
[Stratified random points](https://code.earthengine.google.com/80d1654068551ff059d04f40830b4ba6)

**Comment by ivanzvonkov** [comment]
Next steps:
- CEO labeling project for stratified points ( specifically additional crop points)
- Consider sampling crop points from field boundaries for validation set

**Comment by MsPixels** [comment]
Metrics for the model: 

   "Namibia_North_V3": {
        "params": "https://wandb.ai/nasa-harvest/crop-mask/runs/qzfflhy8",
        "test_metrics": {
            "accuracy": 0.9635,
            "f1_score": 0.1429,
            "precision_score": 0.0833,
            "recall_score": 0.5,
            "roc_auc_score": 0.8404
        },
        "val_metrics": {
            "accuracy": 0.9744,
            "f1_score": 0.1053,
            "precision_score": 0.0556,
            "recall_score": 1.0,
            "roc_auc_score": 0.9834
        }
    },

**Comment by MsPixels** [comment]
[Namibia_North_2020_V3 Map](https://code.earthengine.google.com/a8ef2253efbf7d1bf4f7dca3bf65eb7f?asset=projects%2Fearthengine-legacy%2Fassets%2Fusers%2Fabaansah%2FNamibia_North_2020_V3)

**Comment by MsPixels** [comment]
I also adjusted the threshold [here](https://code.earthengine.google.com/1a68a9ecda9febc88b9b4981c110b403?asset=projects%2Fearthengine-legacy%2Fassets%2Fusers%2Fabaansah%2FNamibia_North_2020_V3). I think it removes some noise

**Comment by hannah-rae** [comment]
@MsPixels initial evaluation: "I think the model does a good job at the west but over-predicts at the east, especially in the flood plains. Some built-up areas were captured as cropland. I'm not super confident about the South. Waiting for confirmation from Christina."

@MsPixels will try the post-classification filtering to see if that reduces some false positive areas (e.g. water/lake shorelines) and we will work with Christina to make final decision.

**Comment by MsPixels** [comment]
[Post-classification](https://code.earthengine.google.com/39282149deafd3baa6edce2b08940b73) filtering @hannah-rae.

**Comment by MsPixels** [comment]
Hello @hannah-rae, after further discussion with Christina and Blake, I generated an NDVI, water and built-up mask to remove most of the noise in the cropland map. The code is [here](https://code.earthengine.google.com/0e37fbc71253eb3165fd7fb41e4f080f)
